shell/common/shell.cc

```diff
@@ -610,6 +610,16 @@ void Shell::OnPlatformViewSetViewportMetrics(const ViewportMetrics& metrics) {
   FML_DCHECK(is_setup_);
   FML_DCHECK(task_runners_.GetPlatformTaskRunner()->RunsTasksOnCurrentThread());
 
+  // This is the formula Android uses.
```

Comment:
Where is this method called from?

On android it looks like the engine method is called directly: https://github.com/flutter/engine/blob/0a2e28d797faf797f348e0338e6536791d279166/shell/platform/android/android_shell_holder.cc#L184

iOS looks like it does the same: https://github.com/flutter/engine/blob/1dcd5f52b5df004c9246d73ee6e0a5e56afdeada/shell/platform/darwin/ios/framework/Source/FlutterEngine.mm#L132

I'm not seeing this code getting run unless I add `platform_view_->SetViewportMetrics(metrics);` to
https://github.com/flutter/engine/blob/2284210f4275b56abb19ecb38665c31b0b935dfc/shell/platform/android/android_shell_holder.cc#L175

Reply:
This is called by the embedders when the tell the PlatformView to update the viewport metrics.  On iOS, this happens in response to a few things in FlutterViewController's lifecycle, and is funnled through here: https://github.com/flutter/engine/blob/master/shell/platform/darwin/ios/framework/Source/FlutterViewController.mm#L674

On Android, that call results from similar things in FlutterView.java which all funnel through https://github.com/flutter/engine/blob/master/shell/platform/android/io/flutter/view/FlutterView.java#L614.

Unless you're writing your own embedding, you should never call this method.  The reason I put the change here is because this is the method where we learn about screen resolution changes, and we want the cache size to be based on a multiple of screen resolution.

It's possible to override that entirely by using the `Skia` platform channel.

Reply:
If you go down the callstack from FlutterView.java you don't get here, you get to the engine call.

https://github.com/flutter/engine/blob/2284210f4275b56abb19ecb38665c31b0b935dfc/shell/platform/android/io/flutter/view/FlutterView.java#L617

https://github.com/flutter/engine/blob/2284210f4275b56abb19ecb38665c31b0b935dfc/shell/platform/android/platform_view_android_jni.cc#L279

https://github.com/flutter/engine/blob/2284210f4275b56abb19ecb38665c31b0b935dfc/shell/platform/android/android_shell_holder.cc#L184

Before this change calling the engine SetViewportMetrics was equivalent to calling the platform view wrapper around it. Its not after this change.

Reply:
Ahh that's unexpected. The other similar methods in the JNI platform view just forward to platform view.  I think it'd be reasonable to change this one to do the same.

Woudl that unstick you?

Reply:
When I add that line the engine crashes in release mode because surface_ seems to be null here:
https://github.com/flutter/engine/blob/2284210f4275b56abb19ecb38665c31b0b935dfc/shell/common/rasterizer.cc#L424

I'm not sure why surface_ is null in release mode but not debug mode.

Reply:
It depends on where you're calling this - the surface may not have been set up yet and perhaps there's some race condition...

Reply:
I mentioned this on https://github.com/flutter/engine/pull/9729, but since the effect of this change seems to have been to reduce the default cache size from 512mb to 24mb should it be rolled back?